### PR TITLE
WorkspaceOps transaction file

### LINKED ISSUE #6251: WorkspaceOps file transaction support

### LINKED ISSUE #6252: WorkspaceOps file sync support

### PR DESCRIPTION
**Pull request by touilleMan**
closes #6251 
closes #6252

### CODE REVIEW

**Comment by mmmarcos** [comment]
@touilleMan you should add the "Closes #6251 #6252 #6253"

File: libparsec/crates/client/src/workspace/transactions/file_operations.rs [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+
+    pub fn rollback(self) {
+        *self.destination_chunks = self.source;
+    }
+
+    pub fn commit(self, chunk_data: &[u8]) {
+        self.destination_chunks[0]
+            .promote_as_block(chunk_data)
+            .expect("chunk is block-compatible");
+    }
```
Comment (vxgmichel) [inline libparsec/crates/client/src/workspace/transactions/file_operations.rs:401]: Maybe move `mem::replace` here to remove rollback method?

Comment (touilleMan) [reply libparsec/crates/client/src/workspace/transactions/file_operations.rs:401]: :+1: fixed

**Comment by touilleMan** [comment]
> @touilleMan you should add the "Closes #6251 #6252 #6253"

#6253 is not implemented, #6252 is only partially implemented (file conflict is not handled)

I mark #6251 #6252 as completed though, and will created dedicated issues for what remains to be implement :+1:

**Pull request merged**